tsl/src/continuous_aggs/create.c

```diff
@@ -1091,6 +1096,33 @@ cagg_query_supported(const Query *query, StringInfo hint, StringInfo detail, con
 	return true; /* Query was OK and is supported */
 }
 
+static inline int64
+get_bucket_width(CAggTimebucketInfo bucket_info)
+{
+	if (bucket_info.bucket_width == BUCKET_WIDTH_VARIABLE)
+	{
+		/* if bucket size is VARIABLE and have month part then pick it for the arithmetic */
+		if (bucket_info.interval->month != 0)
+			return bucket_info.interval->month;
```

Comment:
hmm this seems completely different from the return value in the other codepath i think it would be cleaner if caller does the switch and decides whether he wants microseconds interval or month interval

Reply:
thinking better about it I guess we always can get the microseconds interval for variable buckets, right?? I mean don't need this switch